newIDE/app/src/GameDashboard/GameAnalyticsEvaluator.js

```diff
@@ -0,0 +1,653 @@
+    from300sTo600sPlayersCount: number,
+    from600sTo900sPlayersCount: number,
+    from900sToInfinityPlayersCount: number,
+
+    over0sPlayersPercent: number,
+    over60sPlayersPercent: number,
+    over180sPlayersPercent: number,
+    over300sPlayersPercent: number,
+    over600sPlayersPercent: number,
+    over900sPlayersPercent: number,
+
+    below60sPlayersPercent: number,
+    from60sTo180sPlayersPercent: number,
+    from180sTo300sPlayersPercent: number,
+    from300sTo600sPlayersPercent: number,
+    from600sTo900sPlayersPercent: number,
+    from900sToInfinityPlayersPercent: number,
+  |}[],
+  /**
+   * A funnel of the remaining players after a given played duration.
+   */
+  overPlayedDuration: {| duration: number, playersCount: number |}[],
+|};
+
+const emptyChartData = {
+  overview: {
+    viewersCount: 0,
+    playersCount: 0,
+    bounceRatePercent: 0,
+    meanPlayedDurationInMinutes: 0,
+    nearestToMedianDuration: {
+      playersCount: 0,
+      playersPercent: 0,
+      durationInMinutes: 0,
+    },
+    greaterDurationPlayerSurface: {
+      playersCount: 0,
+      playersPercent: 0,
+      durationInMinutes: 0,
+    },
+  },
+  overTime: [],
+  overPlayedDuration: [],
+};
+
+const durationIndexes: { [string]: number } = {
+  for1Minute: 0,
+  for3Minutes: 1,
+  for5Minutes: 2,
+  for10Minutes: 3,
+  for15Minutes: 4,
+};
+export const durationValues = [1, 3, 5, 10, 15];
+
+/**
+ * Fill the void left by the backend for days without any players.
+ * @param gameMetrics concise game metrics from the backend (today first)
+ * @returns game metrics with a metric for each 364 past days (today first).
+ */
+const fillMissingDays = (
+  gameMetrics: Array<GameMetrics>
+): Array<GameMetrics> => {
+  const filledGameMetrics = [];
+  // TODO In some timezones, it might start the wrong day.
+  let previousMetricDate = addDays(new Date(), 1);
+  for (const metric of gameMetrics) {
+    const metricDate = parseISO(metric.date);
+    // Fill holes
+    while (
+      differenceInCalendarDays(parseISO(metric.date), previousMetricDate) < -1
+    ) {
+      const addedMetricDate = subDays(previousMetricDate, 1);
+      const zeroMetric: GameMetrics = {
+        date: formatISO(addedMetricDate),
+
+        sessions: {
+          d0Sessions: 0,
+          d0SessionsDurationTotal: 0,
+        },
+        players: {
+          d0Players: 0,
+          d0NewPlayers: 0,
+          d0PlayersBelow60s: 0,
+          d0PlayersBelow180s: 0,
+          d0PlayersBelow300s: 0,
+          d0PlayersBelow600s: 0,
+          d0PlayersBelow900s: 0,
+        },
+        retention: null,
+      };
+      filledGameMetrics.push(zeroMetric);
+      previousMetricDate = addedMetricDate;
+    }
+    filledGameMetrics.push(metric);
+    previousMetricDate = metricDate;
+  }
+  // Fill to one year
+  while (filledGameMetrics.length < daysShownForYear) {
+    const addedMetricDate = subDays(previousMetricDate, 1);
+    const zeroMetric: GameMetrics = {
```

Comment:
this can probably be defined outside of the function `fillMissingDays()` as you're using it twice.
Either a function `getZeroMetric(date)` or just an object defined without the date that you spread here?